### PR TITLE
fix (platform-setup): Prevents special regex characters from interferring with database configuration

### PR DESCRIPTION
**Pull request by domtoupin**
While trying to setup Bonita using a PostgreSQL database, I came across the following error:

```java
------------------------------------------------------
Initializing and configuring Bonita BPM WildFly bundle
------------------------------------------------------
 ____              _ _          ____  _____  __  __
|  _ \            (_) |        |  _ \|  __ \|  \/  |
| |_) | ___  _ __  _| |_ __ _  | |_) | |__) | \  / |
|  _ < / _ \| '_ \| | __/ _` | |  _ <|  ___/| |\/| |
| |_) | (_) | | | | | || (_| | | |_) | |    | |  | |
|____/ \___/|_| |_|_|\__\__,_| |____/|_|    |_|  |_|

(platform Setup 7.4.3)

[INFO] configuration for Database vendor: postgres
[INFO] Connected to 'postgres' database with url: 'jdbc:postgresql://localhost:5436/BonitaBPM' with user: 'postgres'
[INFO] Executed SQL script /home/dtoupin/wildfly-DEV/BonitaBPMCommunity-7.4.3-wildfly-10.1.0.Final/setup/platform_conf/sql/postgres/createTables.sql
[INFO] Executed SQL script /home/dtoupin/wildfly-DEV/BonitaBPMCommunity-7.4.3-wildfly-10.1.0.Final/setup/platform_conf/sql/postgres/createQuartzTables.sql
[INFO] Executed SQL script /home/dtoupin/wildfly-DEV/BonitaBPMCommunity-7.4.3-wildfly-10.1.0.Final/setup/platform_conf/sql/postgres/postCreateStructure.sql
[INFO] Executed SQL script /home/dtoupin/wildfly-DEV/BonitaBPMCommunity-7.4.3-wildfly-10.1.0.Final/setup/platform_conf/sql/postgres/initTables.sql
[INFO] Platform created.
[INFO] Database will be initialized with configuration files from folder: platform_conf/initial
[INFO] Initial configuration files successfully pushed to database
[INFO] Wildfly environment detected with root /home/dtoupin/wildfly-DEV/BonitaBPMCommunity-7.4.3-wildfly-10.1.0.Final
[INFO] Running auto-configuration using file /home/dtoupin/wildfly-DEV/BonitaBPMCommunity-7.4.3-wildfly-10.1.0.Final/setup/database.properties
[ERROR] No group 3
[ERROR] You might get more detailed information about the error by setting 'org.bonitasoft.platform.setup.PlatformSetupApplication' log level to 'DEBUG' in file 'logback.xml' and run again
```
The error puzzled me and after enabling the debug logging level, I finally understood what was going on:
```java
[DEBUG] ERROR: 
java.lang.IndexOutOfBoundsException: No group 3
	at java.util.regex.Matcher.start(Matcher.java:375) ~[na:1.8.0_121]
	at java.util.regex.Matcher.appendReplacement(Matcher.java:880) ~[na:1.8.0_121]
	at java.util.regex.Matcher.replaceAll(Matcher.java:955) ~[na:1.8.0_121]
	at java.lang.String.replaceAll(String.java:2223) ~[na:1.8.0_121]
	at org.bonitasoft.platform.setup.command.configure.BundleConfigurator.replace(BundleConfigurator.java:246) ~[platform-setup-7.4.3.jar:na]
	at org.bonitasoft.platform.setup.command.configure.BundleConfigurator.replaceValues(BundleConfigurator.java:151) ~[platform-setup-7.4.3.jar:na]
	at org.bonitasoft.platform.setup.command.configure.WildflyBundleConfigurator.updateStandaloneXmlFile(WildflyBundleConfigurator.java:149) ~[platform-setup-7.4.3.jar:na]
	at org.bonitasoft.platform.setup.command.configure.WildflyBundleConfigurator.configureApplicationServer(WildflyBundleConfigurator.java:69) ~[platform-setup-7.4.3.jar:na]
	at org.bonitasoft.platform.setup.command.configure.ConfigureCommand.execute(ConfigureCommand.java:37) ~[platform-setup-7.4.3.jar:na]
	at org.bonitasoft.platform.setup.PlatformSetupApplication.execute(PlatformSetupApplication.java:105) [platform-setup-7.4.3.jar:na]
	at org.bonitasoft.platform.setup.PlatformSetupApplication.run(PlatformSetupApplication.java:79) [platform-setup-7.4.3.jar:na]
	at org.bonitasoft.platform.setup.PlatformSetupApplication.main(PlatformSetupApplication.java:64) [platform-setup-7.4.3.jar:na]
```
The issue is that my tool-generated password uses a dollar-sign followed by a number (in this case 3) and this has a special meaning to `String::replaceAll`.

My fix therefore protect values coming the database configuration against a misinterpretation.

### CODE REVIEW

**Comment by nchabanoles** [comment]
Hi @domtoupin,

Could you please accept the Contributor License Agreement by clicking the [Details](https://cla-assistant.io/bonitasoft/bonita-engine?pullRequest=266) link in the pending check list.

This will allow us to use your code as part of the product.

thank you very much for your contribution!
Nicolas

**Comment by domtoupin** [comment]
Hi @nchabanoles,

I just accepted the CLA.

Regards,
Dominique

**Comment by nchabanoles** [comment]
@domtoupin sorry, it seems it did not work properly, could you try again please?

many thanks,
Nicolas

**Comment by domtoupin** [comment]
@nchabanoles Did it work this time?

**Comment by nchabanoles** [comment]
Yes, thanks!

We will be able to integrate the changes after performing more tests.

File: platform/platform-setup/src/main/java/org/bonitasoft/platform/setup/command/configure/BundleConfigurator.java [hunk 1]
```diff
@@ -148,7 +148,7 @@ void copyDriverFile(Path srcDriverFile, Path targetDriverFile, String dbVendor)
 
     String replaceValues(String content, Map<String, String> replacementMap) throws PlatformException {
         for (Map.Entry<String, String> entry : replacementMap.entrySet()) {
-            content = replace(content, entry.getKey(), convertWindowsBackslashes(entry.getValue()));
+            content = replace(content, entry.getKey(), entry.getValue());
```
Comment (tbouffard) [inline platform/platform-setup/src/main/java/org/bonitasoft/platform/setup/command/configure/BundleConfigurator.java]: Should not be modified as this is not part of the fix

**Comment by domtoupin** [comment]
Hi @tbouffard 

I am unlikely going to be able to make the requested changes at short term since I am leaving on vacation tomorrow; I might be able to spare some time afterward although I cannot guarantee when exactly...

**Comment by tbouffard** [comment]
Ok thanks @domtoupin for your reply
I am going to integrate your commit today and perform the change by myself.

**Comment by tbouffard** [comment]
So finally, your work is part of the fix which is going to be included in the next 7.6.0 version (internal issue BS-16999)
I am closing this PR

**Comment by domtoupin** [comment]
@tbouffard Thank you, very appreciated!

**Comment by tbouffard** [comment]
@domtoupin we had thanked you for your contribution in the [release notes page](https://documentation.bonitasoft.com/7.6?page=release-notes) of the just released 7.6.0 beta version.

cc @educhastenier @danila-m

**Pull request closed without merging**